### ISSUE TITLE
Issues found in lab

### ISSUE THREAD

**Issue by Eddevinc** [microsoft/MCW-Big-data-analytics-and-visualization#101]
Big data and Visualisation

1. Exercise 1, Task 1, step 2: Screenshot update
2. Exercise 2, Task 1, step 6: name of the file to be uploaded is not mentioned
3. Exercise 2, Task 2, Step 8: Getting errors in Notebook 1,2 and 3
 -Notebook 1: Getting following error in cmd 12
					
![image](https://user-images.githubusercontent.com/32365275/143242112-c8535b61-c0fc-4016-a3f0-1f5def764405.png)

 -Notebook 2: Issue with panda commands, error: name pd is not defined
 -Notebook 3: While running the 3rd notebook. We need to add URI and PAT in one of the code cell. The instruction is not mentioned clearly. Was not able to figure out which URI to be used.
			
4. Exercise 4, Task 1, step 6: Getting following error after clicking on test connection.
   
![image](https://user-images.githubusercontent.com/32365275/143242236-a2f533b1-ef60-4b7c-b546-8dd0c13e0e7f.png)

**Comment by feaselkl** [comment]
Thanks for the issue.  Responses inline.

> Exercise 1, Task 1, step 2: Screenshot update

That is an older screenshot but it still works.  I'll look at getting it updated, though.

> Exercise 2, Task 1, step 6: name of the file to be uploaded is not mentioned

From step 6:
Select Upload File (1) under Create New Table, and then select either select or drag-and-drop the FlightDelaysWithAirportCodes.csv file into the file area (2). Select Create Table with UI (3).

The filename is in the step instructions:  FlightDelaysWithAirportCodes.csv

> -Notebook 1: Getting following error in cmd 12

Looks like you loaded the `FlightWeatherWithAirportCode.csv` file as flight_delays_with_airport_codes.  These three files have different data and different schemas, so choosing a different one will cause an issue.

> -Notebook 2: Issue with panda commands, error: name pd is not defined

Thanks for catching that in notebook 3; I'll add an import statement.

> -Notebook 3: While running the 3rd notebook. We need to add URI and PAT in one of the code cell. The instruction is not mentioned clearly. Was not able to figure out which URI to be used.

The instruction for the URL is in the Serve the Model section:

Wait for both the Status indicator as well as the Production model indicator to read Ready and then copy the Model URL with "Production" in it. This will take 5-10 minutes to complete.

I'll add an additional note there.

> Exercise 4, Task 1, step 6: Getting following error after clicking on test connection.

Do you have a folder C:\Data on the virtual machine, on which you've extracted the files?  The error message is indicating either that the folder does not exist and creating this folder is part of Exercise 3 Task 1.

**Comment by DawnmarieDesJardins** [comment]
Hello! (added Tim for visibility)
@Eddevinc - were you able to complete the lab with @feaselkl suggestions above?

**Comment by Eddevinc** [comment]
Hello @DawnmarieDesJardins and @feaselkl,
I see that the previously mentioned issues by me are working fine now but, now I am encountering new issues which I am mentioning below:

1. Exercise 5, Task 2, step 4: the run is initially in "In Progress" state but later it fails and I cross checked the cluster and it was up and running.
![image](https://user-images.githubusercontent.com/32365275/144045923-3ee813f4-7e3d-409b-806b-cb71d0bf1b82.png)

If I visit the URL mentioned in the error I am redirected to this page, just incase if this info helps
![image](https://user-images.githubusercontent.com/32365275/144046197-7b47378a-a076-462f-8d6e-bf941eb9b22e.png)


2. Exercise 6, task 1, step 1: cells are failing in "Exercise 6 >> 01 Explore Data" notebook

![image](https://user-images.githubusercontent.com/32365275/144046463-7c028e83-dd2b-418e-994d-16286372ad18.png)
![image](https://user-images.githubusercontent.com/32365275/144046507-d1ae9126-cf5d-4705-b62f-75bfdd1b3d9d.png)

**Comment by feaselkl** [comment]
> Exercise 5, Task 2, step 4: the run is initially in "In Progress" state but later it fails and I cross checked the cluster and it was up and running.

The error mentions that no parameters were set.  Can you try again and confirm that you set the windowStart parameter exactly as it is shown?  Just in case, please also make sure to edit the notebook and set your account name in the code block which reads:

```python
accountName = "STORAGE-ACCOUNT-NAME"
containerName = "sparkcontainer"
```

Exercise 6 won't work if Exercise 5 fails because 5 creates the `scoredflights` table.

**Comment by Eddevinc** [comment]
Hi @feaselkl I ran the pipeline again making sure that the windowStart parameter is same as mentioned in lab guide
![image](https://user-images.githubusercontent.com/32365275/144050416-c0542b2e-2a5a-41de-95a2-be4e21e4650f.png)

and the account name in notebook is also set as shown below
![image](https://user-images.githubusercontent.com/32365275/144050815-06fc95e5-efe7-4c65-ac83-ff9d1888d063.png)


Yet the run failed again
![image](https://user-images.githubusercontent.com/32365275/144050712-732ed4a6-54e6-4112-95e8-30f635a5ce5d.png)

**Comment by DawnmarieDesJardins** [comment]
@feaselkl - Hello, happy Friday.  Any update for us?

**Comment by feaselkl** [comment]
Not yet.  I'm going to run through the lab tonight and ensure that everything works.  I'd need more information to determine what the issue is above, but don't know what to ask just yet.

**Comment by feaselkl** [comment]
@Eddevinc the way I was able to replicate your issue was by creating a cluster without the storage account configured.  The error message gives you a URL with run details for the notebook.  Review the `dfDelays = spark.read.csv` command and it should state that it could not find a `sparkcontainer` container.  If so, please delete and re-create the lab cluster using the instructions in Exercise 1 Task 2 Step 5, noting the Spark Config bit:

>Spark Config: Expand Advanced Options and edit the Spark Config by entering the connection information for your Azure Storage account that you copied above in Task 1. This will allow your cluster to access the lab files. Enter the following:
> `spark.hadoop.fs.azure.account.key.<STORAGE_ACCOUNT_NAME>.blob.core.windows.net <ACCESS_KEY>`, where <STORAGE_ACCOUNT_NAME> is your Azure Storage account name, and <ACCESS_KEY> is your storage access key.

**Comment by DawnmarieDesJardins** [comment]
Closing issue as resolved with merge of PR #102
